src/expression.d

```diff
@@ -80,6 +80,43 @@ void emplaceExp(T : UnionExp)(T* p, Expression e)
 }
 
 /*************************************************************
+ * Check if e is a DotVarExp representing a misaligned pointer.
+ * Print error if misaligned pointer and in @safe.
+ * Params:
+ *      sc = scope
+ *      e = expression to check
+ *      msg = error message string
+ * Returns:
+ *      true if error
+ */
+
+bool checkMisalignedPointer(Scope* sc, Expression e, string msg)
+{
+    if (e.op == TOKdotvar)
```

Comment:
This is a prime example of that pattern that should be replaced with:

```D
if (e.op != TOKdotvar)
    return false;
... interesting branch ...
```

Clarifies the charter of the scope and unindents everything that's interesting.